### DISCUSSION TITLE
Avoid errors with empty pipeline

### DISCUSSION THREAD

**Discussion by schmitzli** [nushell/nushell#7531] [Q&A] [answered]
Hey all, many thanks for your continued development on this gem.

I've got a general issue for situations when the pipeline is empty i want to end gracefully.

`help commands | where name =~ 'time' | sort-by category `

with existing results there is no issue, but when there is nothing found there is an error with sort-by.
This is a simple example only, and have this issue with much longer pipelines.
I'm trying to test for no-result and want to stop the pipeline without error. 
Similar maybe as it works with each that simply is not executed.

Thanks for your suggestions

**Comment by fdncred** [comment, marked as answer]
You can try this to see if it works. `do -i` is supposed to ignore errors.

```
do -i { help commands | where name =~ time | sort-by category }
```
and one that doesn't work
```
do -i { help commands | where name =~ blah | sort-by category }
```

**Reply by schmitzli** [reply]
yes, **do -i** or **try** work by suppressing errors, thank you for the quick answer.

`try { help commands | where name =~ blah | sort-by category }`

Would you say there is nothing like 
` ... | if rowcount == 0 { endpipe } else { echo $in } | ....`

**Reply by schmitzli** [reply]
maybe if the following commands are put into the else block and endpipe simply does nothing ?

**Reply by fdncred** [reply]
There is no `rowcount == 0` but there is `is-empty` that you might? be able to make behave.

**Comment by schmitzli** [comment]
that brought me on the right track, thanks

`... | if (echo $in | is-empty) {} else { ...`

when put into an alias

`alias some = if (echo $in | is-empty) {} else  `

can use like, (where the **sort-by** or even the **each $it.name** was broken before with an empty content)

`help commands | where category == defult | some {sort-by command_type | each {|it| echo $it.name } }`
